block/manager.go

```diff
@@ -256,42 +256,30 @@ func (m *Manager) ProcessFraudProof(ctx context.Context, cancel context.CancelFu
 	}
 	defer sub.Cancel()
 
-	// continuously process the fraud proofs received via subscription
-	for {
```

Comment:
[question] What if we get an invalid fraud-proof? Shouldn't we then stay in the for loop?

Reply:
before we had for-loop because we we not fully validating the received fraud proof (meaning `m.executor.VerifyFraudProof` was executed on receiving the fraud proof. Now all the validation is performed prior to gossiping the fraud proofs, hence receiving the first fraud proof (which is always valid) is enough, hence no for-loop. However, this may need to change when we address the halting scenarios (different issue/PR).

Reply:
Mentioned in #933